node/app/query.go

```diff
@@ -66,6 +70,51 @@ func HandleNodeNameQuery(app Application, message []byte) interface{} {
 	return global.Current.NodeName
 }
 
+func HandleSignTransaction(app Application, message []byte) interface{} {
+	var tx action.Transaction
+
+	transaction, transactionErr := serial.Deserialize(message, tx, serial.CLIENT)
+
+	signature := []byte{}
```

Comment:
So, this is returning the signed transaction to the client? That's okay, but I think we discussed having the fullnode itself do the signing and broadcast? I may have been wrong about that. If the fullnode does it, it will have to be async or a goroutine (so that it doesn't take too long and mess up consensus). But there isn't a security issue if you do send it back to the client and then it's the client that waits for the transaction to get committed. That offers some degree of error handling, at the expensive of making the client actions take longer...